src/components/proofing-over-time-report.tsx

```diff
@@ -0,0 +1,188 @@
+import { VNode } from "preact";
+import Markdown from "preact-markdown";
+import { useContext, useRef, useState } from "preact/hooks";
+import * as Plot from "@observablehq/plot";
+import { useQuery } from "preact-fetching";
+import { utcWeek, utcDay } from "d3-time";
+import { mean } from "d3-array";
+import useResizeListener from "../hooks/resize-listener";
+import Accordion from "./accordion";
+import PlotComponent from "./plot";
+import {
+  FunnelMode,
+  ReportFilterContext,
+  Scale,
+  TimeBucket,
+} from "../contexts/report-filter-context";
+import {
+  DailyDropoffsRow,
+  loadData,
+  Step,
+  StepCount,
+  toStepCounts,
+} from "../models/daily-dropoffs-report-data";
+import { formatAsPercent, formatSIDropTrailingZeroes } from "../formats";
+import { useAgencies } from "../contexts/agencies-context";
+
+interface StepCountEntry extends StepCount {
+  date: Date;
+  agency: string;
+  issuer: string;
+  friendlyName: string;
+}
+
+function flatten({
+  data,
+  funnelMode,
+}: {
+  funnelMode: FunnelMode;
+  data: DailyDropoffsRow[];
+}): StepCountEntry[] {
+  const results = data.flatMap((row) => {
+    const { start, agency: rowAgency, issuer, friendly_name: friendlyName } = row;
+
+    return toStepCounts(row, funnelMode).map((stepCount) => ({
+      date: start,
+      issuer,
+      friendlyName,
+      agency: rowAgency,
+      ...stepCount,
+    }));
+  });
+
+  return results;
+}
+
+export default function ProofingOverTimeReport(): VNode {
+  const ref = useRef(null as HTMLDivElement | null);
+  const [width, setWidth] = useState(undefined as number | undefined);
+  const { start, finish, agency, env, funnelMode, scale, byAgency, timeBucket } =
+    useContext(ReportFilterContext);
+
+  const { data } = useQuery(`${start.valueOf()}-${finish.valueOf()}`, () =>
+    loadData(start, finish, env)
+  );
+
+  useResizeListener(() => setWidth(ref.current?.offsetWidth));
+  useAgencies(data);
+
+  const flatSteps = flatten({ data: data || [], funnelMode });
+
+  const lineDeterminer = (() => {
+    if (byAgency) {
```

Comment:
With `byAgency` filtering, could we have some label shown when hovering a line, similar to how it works in the other reports?

Reply:
>My instinct is to save that for a second pass instead of waiting on it? How strongly do you feel?

Not that strongly, especially since `byAgency` isn't really exposed at the moment.